sympy/core/expr.py

```diff
@@ -813,7 +813,7 @@ def _eval_interval(self, x, a, b):
             return B - A
 
         value = B - A
-        if all([S(a).is_comparable, S(b).is_comparable]):
+        if all([a.is_comparable, b.is_comparable]):
```

Comment:
It's about the same for testing two values.But I'll change this one because it looks better. 
```
>>> timeit.timeit('all([S(2).is_comparable,(1+2*I).is_comparable])',number=10000,setup='from sympy import *')
<timeit-src>:2: SyntaxWarning: import * only allowed at module level
0.5940268039703369
>>> timeit.timeit('S(2).is_comparable and (1+2*I).is_comparable',number=10000,setup='from sympy import *')
<timeit-src>:2: SyntaxWarning: import * only allowed at module level
0.5742201805114746
```